pkg/descheduler/node/node.go

```diff
@@ -323,3 +330,18 @@ func PodMatchNodeSelector(pod *v1.Pod, node *v1.Node) bool {
 	}
 	return matches
 }
+
+// podMatchesInterPodAntiAffinity checks if the pod matches the anti-affinity rule
+// of another pod that is already on the given node.
+// If a match is found, it returns true.
+func podMatchesInterPodAntiAffinity(nodeIndexer podutil.GetPodsAssignedToNodeFunc, pod *v1.Pod, node *v1.Node) (bool, error) {
```

Comment:
[NIT] - I think the descheduler terminology here is `violates` anti-affinity, e.g. [the RemovePodsViolatingInterPodAntiAffinity plugin](https://github.com/kubernetes-sigs/descheduler?tab=readme-ov-file#removepodsviolatinginterpodantiaffinity).

I suppose it's not necesarily violating anti-affinity, since it hasn't actually happened yet, so probably not a big deal.  `podWillViolateInterPodAntiAffinity` sounds dumb too.  Maybe leave it how it is now that I think through it.

Reply:
The different naming was intentional! In my interpretation, I see a different angle/perspective/approach between what `NodeFit()` and the plugin each do.

The plugin looks at _all_ nodes, namespaces, and pods for pairs of pods that violate an inter-pod anti-affinity rule. If the Pods violate this rule, one of them is evicted. Whereas `NodeFit()` checks to see if there is a match so that it does not schedule it in the first place. My interpretation of "violating the rule" was that the two Pods with anti-affinity should already be present on the Node, otherwise it's just a match - but it's just an interpretation and maybe it could be worded differently or more clearly !

Reply:
sound reasonable to me!